### PR TITLE
add department and position to user of user find response

### PR DESCRIPTION
**Pull request by MariyaSin**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/701

### CODE REVIEW

**Review by BelehovEgor — CHANGES_REQUESTED** [review]

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 1]
```diff
@@ -19,7 +20,8 @@ public class FindUserCommand : IFindUserCommand
         private readonly IUserInfoMapper _mapper;
         private readonly IUserRepository _repository;
         private readonly ILogger<FindUserCommand> _logger;
-        private readonly IRequestClient<IFindDepartmentUsersRequest> _requestClient;
+        private readonly IRequestClient<IFindDepartmentUsersRequest> _findDepartmentUserRequestClient;
+        private readonly IRequestClient<IGetUsersDepartmentsUsersPositionsRequest> _getUsersDepartmentsUsersPositionsRequestClient;
```
[the remark this answers is not in the document]
Comment (MariyaSin) [reply src/UserService.Business/Commands/User/FindUserCommand.cs:24]: No, in IFindDepartmentUsersRequest i get by department users(only users in one department).
IGetUsersDepartmentsUsersPositionsRequest i get list departments

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 2]
```diff
@@ -51,23 +53,62 @@ private IFindDepartmentUsersResponse GetUserIdsByDepartment(Guid departmentId, i
...
+                {
+                    _logger.LogInformation("Errors while getting users departments and positions." +
+                        "Reason: {Errors}", string.Join('\n', response.Message.Errors));
+                }
+            }
+            catch (Exception exc)
+            {
+                _logger.LogError(exc, errorMessage);
+
+                errors.Add(errorMessage);
```
[the remark this answers is not in the document]
Comment (MariyaSin) [reply src/UserService.Business/Commands/User/FindUserCommand.cs]: done

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 3]
```diff
@@ -78,21 +119,32 @@ public UsersResponse Execute(int skipCount, int takeCount, Guid? departmentId)
...
+                        positions?.UsersPosition?.FirstOrDefault(x => x.UserIds.Contains(dbUser.Id)))));
                 }
             }
             else
             {
                 dbUsers = _repository.Find(skipCount, takeCount, out totalCount);
-            }
 
-            result.TotalCount = totalCount;
+                var departmentsAndPositions = GetUsersDepartmentsUsersPositions(dbUsers.Select(x => x.Id).ToList(), true, true, result.Errors);
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/User/FindUserCommand.cs:137]: remove tab

File: src/UserService.Data/Interfaces/IUserRepository.cs [hunk 4]
```diff
@@ -18,9 +18,9 @@ public interface IUserRepository
 
         DbUser Get(Guid id);
 
-        IEnumerable<DbUser> Get(IEnumerable<Guid> userIds);
+        List<DbUser> Get(IEnumerable<Guid> userIds);
```
Comment (BelehovEgor) [inline src/UserService.Data/Interfaces/IUserRepository.cs:21]: for what?

File: src/UserService.Data/UserRepository.cs [hunk 5]
```diff
@@ -198,11 +198,16 @@ public bool SwitchActiveStatus(Guid userId, bool status)
...
         {
+            if (takeCount <= 0)
+            {
+                throw new BadRequestException("Take count can't be equal or less than 0.");
+            }
+
             totalCount = _provider.Users.Count();
 
-            return _provider.Users.Skip(skipCount * takeCount).Take(takeCount).ToList();
+            return _provider.Users.Skip(skipCount).Take(takeCount).ToList();
```
Comment (BelehovEgor) [inline src/UserService.Data/UserRepository.cs:210]: skipCount * takeCount was right

**Review by BelehovEgor — APPROVED** [review]

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 6]
```diff
@@ -44,30 +46,69 @@ private IFindDepartmentUsersResponse GetUserIdsByDepartment(Guid departmentId, i
...
+                    .GetResponse<IOperationResult<IGetUsersDepartmentsUsersPositionsResponse>>(request)
+                    .Result;
+
+                if (response.Message.IsSuccess)
+                {
+                    return response.Message.Body;
+                }
+                else
+                {
+                    _logger.LogInformation("Errors while getting users departments and positions." +
```
Comment (djino97) [inline src/UserService.Business/Commands/User/FindUserCommand.cs:78]: LogWarning

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 7]
```diff
@@ -29,11 +31,11 @@ private IFindDepartmentUsersResponse GetUserIdsByDepartment(Guid departmentId, i
...
-                var response = _requestClient.GetResponse<IOperationResult<IFindDepartmentUsersResponse>>(request, timeout: RequestTimeout.Default).Result;
+                var response = _findDepartmentUserRequestClient.GetResponse<IOperationResult<IFindDepartmentUsersResponse>>(request, timeout: RequestTimeout.Default).Result;
 
                 if (response.Message.IsSuccess)
                 {
-                    users = response.Message.Body;
+                    return response.Message.Body;
                 }
                 else
                 {
```
Comment (djino97) [inline src/UserService.Business/Commands/User/FindUserCommand.cs:41]: fix -- LogWarning

File: src/UserService.Mappers/Models/UserInfoMapper.cs [hunk 8]
```diff
@@ -8,28 +9,39 @@ namespace LT.DigitalOffice.UserService.Mappers.Models
...
+            response.LastName = dbUser.LastName;
+            response.MiddleName = dbUser.MiddleName;
+            response.Gender = (UserGender)dbUser.Gender;
+            response.DateOfBirth = dbUser.DateOfBirth?.ToShortDateString();
+            response.City = dbUser.City;
+            response.StartWorkingAt = dbUser.DateOfBirth?.ToShortDateString();
+            response.About = dbUser.About;
+            response.IsAdmin = dbUser.IsAdmin;
+            response.Status = (UserStatus)dbUser.Status;
+            response.Rate = dbUser.Rate;
```
Comment (djino97) [inline src/UserService.Mappers/Models/UserInfoMapper.cs:42]: add isActive

**Review by djino97 — APPROVED** [review]

File: src/UserService.Business/Commands/User/FindUserCommand.cs (hunk 6 above)
Comment (MariyaSin) [reply src/UserService.Business/Commands/User/FindUserCommand.cs]: done

File: src/UserService.Business/Commands/User/FindUserCommand.cs (hunk 7 above)
Comment (MariyaSin) [reply src/UserService.Business/Commands/User/FindUserCommand.cs:41]: done

**Review by TIove — APPROVED** [review]

**Pull request merged**